doc/md/versioned/troubleshoot.mdx

```diff
@@ -0,0 +1,289 @@
+
+```text
+$ atlas --env local migrate status
+Migration Status: PENDING
+  -- Current Version: 2 (2 statements applied)
+  -- Next Version:    2 (1 statements left)
+  -- Executed Files:  2 (last one partially)
+  -- Pending Files:   2
+
+Last migration attempt had errors:
+  -- SQL:   INSERT INTO `user_groups` (`user_id`, `group_id`) VALUES     -- Founders     (2, 1), (3, 1),     -- Seniors     (1, 2), (2, 2), (3, 2),     -- Constraint error (adding masseelch twice to seniors)     (1, 2);
+// highlight-next-line-error-message
+  -- ERROR: Error 1062: Duplicate entry '1-2' for key 'user_groups.PRIMARY'
+```
+
+</TabItem>
+</Tabs>
+
+## Fixing Migration Failures
+
+In our experience, the most common failures are introduced by one of the following issues:
+
+- **Syntax Error**: One SQL statement contained a syntax error. Since Atlas provides a way to [lint](/versioned/lint) a
+migration directory, this case happens due to a misconfiguration, e.g. not executing the linter before applying the
+migration or applying against the wrong dialect (like SQLite files against MySQL).
+- **Schema Dependant Change**: One SQL statement was incorrect in schema context, e.g. adding a table that does already
+exist. This case is usually caused by a schema drift, e.g. because someone manually manipulated the database schema.
+- **Data Dependant Change**: One SQL statement contained a data dependant change. If your migration adds a constraint
+like `NOT NULL` and the column does contain `NULL` values, the migration will throw an error. Though Atlas' linting
+will warn if such a change is detected, this is still a possible scenario that could break migration execution.
+- **Connection Loss**: For a number of reasons the connection to the database could be interrupted, e.g. due to a
+network failure or database crash.
+
+### Connection Loss Failures
+
+Just re-try migration execution. Atlas is smart enough to figure out where to proceed execution. For very rare cases,
+where when working without a transaction (or after an implicit commit) a connection loss occurs between an SQL statement
+and the entry in Atlas schema history table, you might need to manually revert the last applied statement because in
+those cases Atlas will continue applying one statement early.
+
+### Syntax Errors
+
+Fix the migration file and re-try migration execution. Atlas is smart enough to continue the execution starting with the
+statement following the last successfully applied one.
+
+### DDL Semantics
+
+Creating or editing schema resources can result in migration failure, e.g. if you are trying to create a table that does
+already exist or to modify a table that does not exist. The cause for this is either an incorrect migration file or a
+schema drift, e.g. if someone dropped, created or modified a resource.
+
+#### Option 1: Incorrect Migration File
+
+Solution can be simple: fix the migration file and re-apply the migration. For cases with multiple deployments or no
```

Comment:
```suggestion
In case the migration script is incorrect, simply fix the migration file and re-apply the migration. Do not forget to update the migration directory checksum afterwards by running:
```shell
atlas migrate hash --force
```

For cases with multiple deployments or no
```